deployment/src/main/resources/templates/ConfigSource.qute

```diff
@@ -0,0 +1,13 @@
+package {packageName};
+
+import io.quarkiverse.asyncapi.config.AsyncConfigSource;
+import java.io.IOException;
+import io.quarkiverse.asyncapi.config.ObjectMapperFactory;
+import io.quarkiverse.asyncapi.config.AsyncAPIExtension;
+import com.asyncapi.v2.model.AsyncAPI;
+
+public class {id}ConfigSource extends AsyncConfigSource {
```

Comment:
Instead of using Qute, how about generating these classes during build time using Gizmo?

Reply:
Is isnt qute the one template language recommended by quarkus. 
Again, here I follow the same approach than for OpenAPI generator, using qute during codegen phase.

Reply:
Gizmo isn't a template language, but a library (also used in Quarkus core) to generate classes during build time. That would be less error-prone than using a template. 

But that's a matter of choice, not forcing anything here

Reply:
We are using that approach for Kogito, but rather than Gizmo, we use https://github.com/javaparser/javaparser
The decision to use a template during CodeGen is to intentionally mimic the approach taken in the OpenAPI quarkiverse (I was actually trying to avoid a discussion by doing that ;)). It actually took me more time to make this work that it would have took me using build stept and ast, but as explained before, when we generate the model classes, this will be needed before build step (for people to write code, like in the OpenAPI case).

Reply:
You can also check out  https://github.com/forge/roaster, which is more powerful than JavaParser 😉

Reply:
hmmm we have plans to remove the template engine from the openapi-generator extension once we change the inner generator to Apicurio. 🤔

Reply:
@ricardozanini @gastaldi The point is not using a template, gizmo, java parser or roaster. The point is to use build or generated-source maven goal, so the generated code is avalable earlier (and compile automatically by Maven) so it does not have to be programatically compiled during build . 
Since we need an alpha version of this extension for kogito, I will go with what I did and change later if needed. Later, we will also probably have to change the Async parser dependency by the smallrye one, when evolved.

Reply:
Yeah, I take back what I said. We must use codegen at least for OpenAPI because users would use the generated classes in their project. Templates are the way to go :)